In [1]:
import aad
from time import time

start = time()
driver = aad.AwesomeArrayDriver()
end = time()
print(end-start)

0.10584044456481934


In [2]:
start = time()
for row in range(aad.SR_WORD_SIZE):
    for col in range(aad.SR_WORD_SIZE):
        driver.set(col=col, row=row, bar=False)
        driver.set(col=col, row=row, bar=True)
end = time()
print(end-start)

396.10384702682495


#### Before vs After moving shift registers fill from Python to C

Before (Test 1): 858.3955516815186  
Before (Test 2): 856.1749730110168  
========================  
After (Test 1): 394.1490559577942  
After (Test 2): 396.10384702682495  